packages/back-end/src/events/handlers/slack/handleFeatureEventForSlack.ts

```diff
@@ -0,0 +1,312 @@
+import uniq from "lodash/uniq";
+import { KnownBlock } from "@slack/web-api";
+import {
+  FeatureCreatedNotificationEvent,
+  FeatureDeletedNotificationEvent,
+  FeatureUpdatedNotificationEvent,
+} from "../../base-events";
+import { getSlackIntegrationsForFilters } from "../../../models/SlackIntegrationModel";
+import { FeatureEnvironment } from "../../../../types/feature";
+import { SlackIntegrationInterface } from "../../../../types/slack-integration";
+import { logger } from "../../../util/logger";
+import { APP_ORIGIN } from "../../../util/secrets";
+import { sendSlackMessage, SlackMessage } from "./slack-event-handler-utils";
+
+type HandleFeatureEventOptions = {
+  organizationId: string;
+  eventId: string;
+  featureEvent:
+    | FeatureCreatedNotificationEvent
+    | FeatureUpdatedNotificationEvent
+    | FeatureDeletedNotificationEvent;
+};
+export const handleFeatureEventForSlack = async ({
+  organizationId,
+  eventId,
+  featureEvent,
+}: HandleFeatureEventOptions) => {
+  // Build filtering query and get relevant SlackIntegration records
+  const tags = getTagsForFeatureEvent(featureEvent);
+  const environments = getEnvironmentsForFeatureEvent(featureEvent);
+  const projects = getProjectsForFeatureEvent(featureEvent);
+
+  const slackIntegrations = await getSlackIntegrationsForFilters({
+    organizationId,
+    eventName: featureEvent.event,
+    tags,
+    environments,
+    projects,
+  });
```

Comment:
Here we build the filters for our query. See the method `getSlackIntegrationsForFilters` below for more info about how it works.